lte/gateway/c/core/oai/test/amf/test_amf.cpp

```diff
@@ -298,6 +304,23 @@ TEST(test_amf_nas5g_pkt_process, test_amf_pdu_sess_est_req_type2_msg) {
   sm_free_protocol_configuration_options(&pco_resp);
 }
 
+TEST(test_amf_nas5g_pkt_process, test_amf_pdu_sess_est_req_type3_msg) {
+  NAS5GPktSnapShot nas5g_pkt_snap;
+  ULNASTransportMsg pdu_sess_est_req;
+  bool decode_res      = false;
+  uint8_t buffer[1024] = {};
+  uint16_t buf_len     = 1024;
+
+  uint32_t len = nas5g_pkt_snap.get_pdu_session_est_type3_len();
+
+  /* Check if uplink pdu packet is parsed properly */
+  memset(&pdu_sess_est_req, 0, sizeof(ULNASTransportMsg));
+  decode_res = decode_ul_nas_transport_msg(
+      &pdu_sess_est_req, nas5g_pkt_snap.pdu_session_est_req_type3, len);
+
+  EXPECT_EQ(decode_res, true);
```

Comment:
this does not do much. you should compare against the expected fields.

Reply:
@VinayKumar1591 can you submit another set of PRs that actually perform both encoding and decoding?